packages/core/bookingLimits/getBusyTimesFromLimits.ts

```diff
@@ -0,0 +1,254 @@
+import type { Dayjs } from "@calcom/dayjs";
+import dayjs from "@calcom/dayjs";
+import { descendingLimitKeys, intervalLimitKeyToUnit } from "@calcom/lib/intervalLimit";
+import { checkBookingLimit } from "@calcom/lib/server";
+import { performance } from "@calcom/lib/server/perfObserver";
+import { getTotalBookingDuration } from "@calcom/lib/server/queries";
+import { BookingRepository } from "@calcom/lib/server/repository/booking";
+import type { EventBusyDetails, IntervalLimit } from "@calcom/types/Calendar";
+
+import { getStartEndDateforLimitCheck } from "../getBusyTimes";
+import type { EventType } from "../getUserAvailability";
+import { getPeriodStartDatesBetween } from "../getUserAvailability";
+import monitorCallbackAsync from "../sentryWrapper";
+import LimitManager from "./limitManager";
+
+export const getBusyTimesFromLimits = async (
```

Comment:
all function are copied from getUserAvailability, only https://github.com/calcom/cal.com/pull/16614/files#diff-aa4a69fe03b40580e246864cb9e2479ec6eb64e787f136354bf8bd5b4e2722e2R211 is new